### DISCUSSION TITLE
singleton unit testing

### DISCUSSION THREAD

**Discussion by ranran2121** [prisma/prisma#19254] [Q&A]
Hi, I followed the documentation [https://www.prisma.io/docs/guides/testing/unit-testing](here) in order to unit test using the singleton method
However the expected value results as "null" and I am still not able to figure out the reason why
these are the files with the code
client.ts
```
import { PrismaClient } from "@prisma/client";

const prisma = new PrismaClient();
export default prisma;
```

singleton.ts
```
import { PrismaClient } from "@prisma/client";
import { mockDeep, mockReset, DeepMockProxy } from "jest-mock-extended";

import prisma from "./client";

jest.mock("./client", () => ({
  __esModule: true,
  default: mockDeep<PrismaClient>(),
}));

beforeEach(() => {
  mockReset(prismaMock);
});

export const prismaMock = prisma as unknown as DeepMockProxy<PrismaClient>;
```

comment_controller_test.ts
```
import {  viewComment } from "../../controller/comments";
import { prismaMock } from "../../../src/prisma/singleton";

describe("View Comment /", () => {
  describe("comment/:id", () => {
    it("should views details of single comment ", async () => {
      const comment = {
        id: 100,
        description: "Richi Rich",
        rate: 4,
        userId: 1,
        resourceId: 1,
        createdAt: new Date("2023-03-29"),
        updatedAt: new Date("2023-03-29"),
      };

      prismaMock.comment.findUnique.mockResolvedValueOnce(comment);

      await expect(viewComment(comment.id)).resolves.toEqual(comment);
    });
  });
});
```

this is the result after running the file
```
expect(received).resolves.toEqual(expected) // deep equality

    Expected: {"createdAt": 2023-03-29T00:00:00.000Z, "description": "Richi Rich", "id": 100, "rate": 4, "resourceId": 1, 
      "updatedAt": 2023-03-29T00:00:00.000Z, "userId": 1}
    Received: null
```

Any help is appreciated

**Comment by ludralph** [comment]
Hi @ranran2121 👋 

Can you please share a reproduction in a repository so I can investigate further?

**Reply by ranran2121** [reply]
I have reproduced this mini-repo....let me know if it works because the real one is quite more complex :(

https://codesandbox.io/s/js-playground-forked-m52v61?file=/src/prisma/singleton.ts

**Reply by ludralph** [reply]
Thanks for sharing, I'll take a look now.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
